safe_transaction_service/history/tests/test_commands.py

```diff
@@ -443,3 +452,45 @@ def test_check_chainid_matches(self, get_chain_id_mock: MagicMock):
         buf = StringIO()
         call_command(command, stdout=buf)
         self.assertIn("EthereumRPC chainId 1 looks good", buf.getvalue())
+
+    def test_check_index_problems(self):
+        command = "check_index_problems"
+        buf = StringIO()
+        # Test empty safecontracts
+        call_command(command, stdout=buf)
+        self.assertIn("Database haven't any address to be checked", buf.getvalue())
+        owner = Account.create()
+        safe = self.deploy_test_safe(
+            number_owners=1,
+            threshold=1,
+            owners=[owner.address],
+            initial_funding_wei=1000,
+        )
+        SafeContractFactory(address=safe.address)
+        last_status = SafeLastStatusFactory(nonce=0, address=safe.address)
+        buf = StringIO()
+        with mock.patch(
+            "safe_transaction_service.history.management.commands.check_index_problems.settings.ETH_L2_NETWORK",
+            return_value=True,
+        ):
+            call_command(command, stdout=buf)
+            self.assertIn("Database haven't any address to be checked", buf.getvalue())
+
+            data = b""
+            value = 122
+            to = Account.create().address
+            multisig_tx = safe.build_multisig_tx(to, value, data)
+            multisig_tx.sign(owner.key)
+            tx_hash, _ = multisig_tx.execute(self.ethereum_test_account.key)
+            SafeLastStatus.objects.filter(address=safe.address).update(nonce=1)
+            self.assertEqual(InternalTxDecoded.objects.count(), 0)
+            buf = StringIO()
+            call_command(command, stdout=buf)
+            self.assertIn(
+                f"Safe={safe.address} is corrupted, has some old transactions missing",
+                buf.getvalue(),
+            )
+            # Should be 1 InternalTxDecoded
+            self.assertEqual(InternalTxDecoded.objects.count(), 1)
```

Comment:
@Uxio0  Check index problems found an issue and trigger the reindex then InternalTxDecoded should have 1 transaction taken from blockchain but the model is empty. The transaction is captured if this reindex https://github.com/safe-global/safe-transaction-service/blob/master/safe_transaction_service/history/management/commands/check_index_problems.py#L132 start in the previous block where the issue was founded. 
What do you think?

Reply:
So you mean that `reindex_master_copies` only works when setting one block less that the one where the tx is missing?

Reply:
It's right, and I'm concerning about the check index problems command can even jump the transactions missing because this.

Reply:
Yes it should 👍🏻